components/resolved_ts/src/endpoint.rs

```diff
@@ -78,78 +80,108 @@ impl ObserveRegion {
 
     fn track_change_log(&mut self, change_logs: &[ChangeLog]) -> Result<()> {
         match self.resolver_status {
-            ResolverStatus::Pending { ref mut locks, .. } => {
+            ResolverStatus::Pending {
+                ref mut locks,
```

Comment:
If you match it using `&mut self.resolver_status` then you don't need to write `ref mut` everywhere.